webpack/ForemanTasks/Components/TasksTable/TasksTableSelectors.js

```diff
@@ -27,18 +28,20 @@ export const selectSelectedRows = state =>
 export const selectModalStatus = state =>
   selectTasksTableQuery(state).modalStatus || CLOSED;
 
-export const selectResults = state => {
-  const { results } = selectTasksTableContent(state);
-  if (!results) return [];
-  return results.map(result => ({
-    ...result,
-    action: result.action || result.label.replace(/::/g, ' '),
-    username: result.username || '',
-    state: result.state + (result.frozen ? ` ${__('Disabled')}` : ''),
-    duration: getDuration(result.started_at, result.ended_at),
-    availableActions: result.available_actions,
-  }));
-};
+export const selectResults = createSelector(
+  selectTasksTableContent,
+  ({ results }) =>
+    results.length
+      ? results.map(result => ({
```

Comment:
Is this necessary? `[].map(resutl => $something)` is still `[]`. Or am I missing something and `results` can be something that cannot be `map`ped over?